### PR TITLE
xim context creation

### PR DESCRIPTION
**Pull request by duarm**
just creates an IM context to be used later, this shouldn't affect anything aside from requiring

#include <xcb-imdkit/imclient.h>

before including libgwater_xcb.h, is this desirable? or I missed something the way the headers are handled?
the context is only put to use when calling xcb_xim_open()

### CODE REVIEW

**Comment by sardemff7** [comment]
First: Reading the code in its current state, I see no reason to include it in libgwater  .
It doesn’t interact with the main loop and you only need an `xcb_connection_t` which is provided already.  
If it needs to be created in a special way at a special moment that lies in libgwater, I think the `_for_connection` variant can be used, with an off-loaded `xcb_connection_t`

TL;DR: it doesn’t belong here, IMO, so I will close this MR

Now, thanks for the effort, and I still have a review comment: when you contribute to a project, please, follow the style that is already in place. Here, specifically, no “long lines wrapping” and the `{` on their own line. :slightly_smiling_face:

**Comment by duarm** [comment]
ok, that method wrapped two times on my editor, thought it was harmless, thanks
